### PR TITLE
feat: Add `filter` flag to `tables` command

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/17524

### CODE REVIEW

**Review by bbernays — APPROVED** [review]

File: .github/workflows/cli.yml [hunk 1]
```diff
@@ -49,7 +49,11 @@ jobs:
       - name: Build
         run: go build .
       - name: Test
-        run: make test
+        if: matrix.os == 'ubuntu-latest'
+        run: go test -v -race ./...
+      - name: Test
+        if: matrix.os != 'ubuntu-latest'
+        run: go test -v ./...
```
Comment (candiduslynx) [inline .github/workflows/cli.yml:56]: I'm worried about this change. Why do we need it?

File: cli/cmd/addon_publish_test.go [hunk 2]
```diff
@@ -91,9 +89,7 @@ func TestAddonPublishEmbedded(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if diff := cmp.Diff(wantCalls, gotCalls); diff != "" {
-		t.Fatalf("mismatch (-want +got):\n%s", diff)
-	}
+	require.Empty(t, cmp.Diff(wantCalls, gotCalls), "unexpected calls to the server")
```
Comment (candiduslynx) [inline cli/cmd/addon_publish_test.go:92]: Please, move refactoring tests to a separate pr

File: cli/cmd/tables.go [hunk 3]
```diff
@@ -34,6 +36,7 @@ func NewCmdTables() *cobra.Command {
 	}
 	cmd.Flags().String("output-dir", "cq-docs", "Base output directory for generated files")
 	cmd.Flags().String("format", "json", "Output format. One of: json, markdown")
+	cmd.Flags().String("filter", "all", "Filter tables. One of: all, spec")
```
Comment (candiduslynx) [inline cli/cmd/tables.go:39]: Maybe make default empty?

Comment (erezrokah) [reply cli/cmd/tables.go:39]: I think `all` conveys better what would be the output the default behavior, WDYT?

Comment (candiduslynx) [reply cli/cmd/tables.go:39]: IMO `--filter` is a restrictive option, so, it it's empty/not specified, `allow all` is implied.
So I'd prefer saying that the default is `""`

Comment (erezrokah) [reply cli/cmd/tables.go:39]: I think from a user perspective the behavior is the same. If they don't specify the flag it will return all tables. Also specifying `--filter all` seem nicer than? Otherwise it looks like a boolean flag I think
